@xen-orchestra/lite/src/views/host/HostConsoleView.vue

```diff
@@ -0,0 +1,190 @@
+<template>
+  <div :class="{ 'no-ui': !uiStore.hasUi }" class="host-console-view">
+    <div v-if="hasError">{{ $t('error-occurred') }}</div>
+    <UiSpinner v-else-if="!isReady" class="spinner" />
+    <UiStatusPanel v-else-if="!isHostRunning" :image-source="monitor" :title="$t('power-on-host-for-console')" />
+    <template v-else-if="host && hostConsole">
+      <MenuList horizontal>
+        <MenuItem v-if="uiStore.hasUi" :icon="faArrowUpRightFromSquare" @click="openInNewTab">
+          {{ $t('open-console-in-new-tab') }}
+        </MenuItem>
+        <MenuItem
+          :icon="uiStore.hasUi ? faUpRightAndDownLeftFromCenter : faDownLeftAndUpRightToCenter"
+          @click="toggleFullScreen"
+        >
+          {{ $t(uiStore.hasUi ? 'fullscreen' : 'fullscreen-leave') }}
+        </MenuItem>
+        <MenuItem :disabled="!consoleElement" :icon="faKeyboard" @click="sendCtrlAltDel">
+          {{ $t('send-ctrl-alt-del') }}
+        </MenuItem>
+      </MenuList>
+      <RemoteConsole
+        ref="consoleElement"
+        :is-console-available="isConsoleAvailable"
+        :location="hostConsole.location"
+        class="remote-console"
+      />
+    </template>
+  </div>
+</template>
+
+<script lang="ts" setup>
+import monitor from '@/assets/monitor.svg'
+import RemoteConsole from '@/components/RemoteConsole.vue'
+import UiSpinner from '@/components/ui/UiSpinner.vue'
+import UiStatusPanel from '@/components/ui/UiStatusPanel.vue'
+import { isVmOperationPending } from '@/libs/vm'
+import { VM_OPERATION } from '@/libs/xen-api/xen-api.enums'
+import type { XenApiHost } from '@/libs/xen-api/xen-api.types'
+import { usePageTitleStore } from '@/stores/page-title.store'
+import { useConsoleStore } from '@/stores/xen-api/console.store'
+import { useControlDomainStore } from '@/stores/xen-api/control-domain.store'
+import { useHostStore } from '@/stores/xen-api/host.store'
+import MenuItem from '@core/components/menu/MenuItem.vue'
+import MenuList from '@core/components/menu/MenuList.vue'
+import { useUiStore } from '@core/stores/ui.store'
+import {
+  faArrowUpRightFromSquare,
+  faDownLeftAndUpRightToCenter,
+  faKeyboard,
+  faUpRightAndDownLeftFromCenter,
+} from '@fortawesome/free-solid-svg-icons'
+import { computed, ref } from 'vue'
+import { useI18n } from 'vue-i18n'
+import { useRoute, useRouter } from 'vue-router'
+
+const STOP_OPERATIONS = [
+  VM_OPERATION.SHUTDOWN,
+  VM_OPERATION.CLEAN_SHUTDOWN,
+  VM_OPERATION.HARD_SHUTDOWN,
+  VM_OPERATION.CLEAN_REBOOT,
+  VM_OPERATION.HARD_REBOOT,
+  VM_OPERATION.PAUSE,
+  VM_OPERATION.SUSPEND,
+]
+
+usePageTitleStore().setTitle(useI18n().t('console'))
+
+const router = useRouter()
+const route = useRoute()
+const uiStore = useUiStore()
+
+const {
+  isReady: isHostReady,
+  getByUuid: getHostByUuid,
+  hasError: hasHostError,
+  runningHosts: getRunningHosts,
+} = useHostStore().subscribe()
+const { records: getRecords } = useControlDomainStore().subscribe()
```

Comment:
To add better readability, what do you think of renaming `getRecords`?
```suggestion
const { records: controlDomains } = useControlDomainStore().subscribe()
```

Reply:
FYI, we generally use the prefixes `get/set` for functions
```js
const foo = getFoo()
```